modules/svg/image_loader_svg.cpp

```diff
@@ -47,7 +46,57 @@ SVGRasterizer::~SVGRasterizer() {
 
 SVGRasterizer ImageLoaderSVG::rasterizer;
 
-Error ImageLoaderSVG::_create_image(Ref<Image> p_image, const PoolVector<uint8_t> *p_data, float p_scale, bool upsample) {
+inline Color nsvg_col_to_godot_color(int p_hex) {
+	//swaps the color from ABGR to RGBA (nsvg is usging ABGR)
+
+	return Color::hex(
+			(p_hex & 0xFF) << 24 | //red to << 24
+			(p_hex & (0xFF << 8)) << 8 | //green to << 16
+			(p_hex & (0xFF << 16)) >> 8 | //blue to << 8
+			0xFF //alpha to << 0 and set to 255
+			);
+}
+inline void change_nsvg_paint_color(NSVGpaint *p_paint, const Color p_old, const Color p_new) {
```

Comment:
also here you can pass unsigned ints instead of colors, so before calling convert the colors using to_RGBA32(), and thats it, you dont need to convert back and forth.

Reply:
you can then get rid of `nsvg_col_to_godot_color`

Reply:
that is a really good hint. I was adding the to_ABGR32 (can us to_RGBA32... because of the swapped color channel) function at the end and didn't consider a simpler implementation based on it.
Just was happy that it finally worked.
before that I was still trying to fighting against the weird order of color channels of nanosvg colors.

Reply:
this is still valid

Reply:
yea should definitely be added with your version. it's much nicer!

Reply:
oh but I'm not sure if nsvg guarantees 1111 1111 for the alpha channel. I think that was another issue. it is more or less random...
so for comparing I need to only compare the last 24 bytes.
by shifting to the left it always fills up with 0000 0000 is that correct?
If so I could shift both:
`0101 0101 0010 0000 0011 1111 0110 1010`
`1111 1111 0010 0000 0011 1111 0110 1010`
->
`0010 0000 0011 1111 0110 1010 0000 0000`
`0010 0000 0011 1111 0110 1010 0000 0000`
they are the same

Reply:
just tested.
and comparing without shifting does not work.
I will try with shifting and than research if shifting always adds 0000 0000  and if so it should work.

Reply:
Okay with shifting it works !

Reply:
you can also use a bitmask, like ``` uint32_t masked = uintcolor & 0x00FFFFFF ```